src/views/List.jsx

```diff
@@ -1,16 +1,47 @@
 import { ListItem } from '../components';
+import { useState } from 'react';
 
 /** List component that displays items in a user's shopping cart  */
 export function List({ data }) {
+	const [filteredList, setFilteredList] = useState([]);
+	const [filterInput, setFilterInput] = useState('');
+
+	/* Use handler to change the state of filterInput  and convert all items to lowercase to facilitate a more thorough search */
+	const handleInput = (event) => {
+		const value = event.target.value.toLowerCase();
+		setFilterInput(value);
+		setFilteredList(
+			data.filter((item) => {
+				return item.name && item.name.toLowerCase().includes(value);
+			}),
+		);
+	};
+
 	return (
 		<>
 			<p>
 				Hello from the <code>/list</code> page!
 			</p>
+
+			<form>
+				<label htmlFor="list-filter">Filter items</label>
+				<br />
+				<input
+					id="list-filter"
+					type="search"
```

Comment:
So your use of the `type="search"` attribute indirectly gives you the "clear" button mentioned by the acceptance criteria, in Google Chrome, but it might not be the same in other browsers. I'd suggest trying it in Firefox, since I'm seeing in the "Can I Use" reference that Firefox might be different from other browsers: https://caniuse.com/input-search

Implementing the "clear" button in React would require that you take control of the `value` prop on the search input.